tests/Tokenizer/Transformer/DisjunctiveNormalFormTypeParenthesisTransformerTest.php

```diff
@@ -324,4 +324,36 @@ public function __invoke((Foo&Bar)|string $a): string
             ],
         ];
     }
+
+    /**
+     * @param array<int, int> $expectedTokens
+     *
+     * @dataProvider provideProcess83Cases
+     *
+     * @requires PHP 8.3
+     */
+    public function testProcess83(string $source, array $expectedTokens): void
+    {
+        $this->doTest($source, $expectedTokens);
+    }
+
+    public static function provideProcess83Cases(): iterable
+    {
+        yield 'typed const alternate types' => [
```

Comment:
Since whole transformer is about DNF I think we can skip this info in the case's description:
```suggestion
        yield 'typed const' => [
```
Existing cases are named after DNF's place of usage, like "constructor property promotion" or "arrow functions".

ℹ️ Same for test cases in other transformers ("alternate types" is used everywhere, even if there's only intersection, so it does not represent actual cases).

Reply:
Unresolved, because I asked for improving yields in other transformers too 😉.

Reply:
I didn't resolved it, I just commited your suggestion, maybe github makes it resolved automatically when you accept a commit suggestion? hmmm

Reply:
Yeah, most probably. I just wanted to keep it until it's confirmed, nothing personal 😅

Reply:
@SpacePossum are you going to fix other test cases' names, as pointed [above](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7055#discussion_r1230230436)?

> ℹ️ Same for test cases in other transformers ("alternate types" is used everywhere, even if there's only intersection, so it does not represent actual cases).

Reply:
I've update the test descriptions, please let me know if any are still off

Reply:
I think cases in `TypeIntersectionTransformerTest` use DNF types rather than "alternate types"... Cosmetics, but still 😉.

Reply:
>I think cases in TypeIntersectionTransformerTest use DNF types

likely as all of `TypeIntersectionTransformerTest` is about DNF handling, 

> rather than "alternate types"

that are these 3:

```
        yield 'typed const alternate types 1' => [            
            '<?php class Foo { const (A&B)|Z Bar = 1;}',

        yield 'typed const alternate types 2' => [
            '<?php class Foo { const Z|(A&B) Bar = 1;}',

        yield 'typed const alternate types 3' => [
            '<?php class Foo { const Z|(A&B)|X Bar = 1;}',
```

all 3 variants are there to test:
-  `)` followed by `|`
- `(` with previous `|`
- `(`  with previous `|` and `)` followed by `|`

so all alternate type?
Again happy to put any other text in there but I don't understand what is expected

Reply:
DNF type is specific type that mixes intersection with union (alternation), so yeah, technically DNF is an union (alternation), but I believe it should be explicitly stated that these cases operate on DNF types. Detail, but it triggers me 😅.

Reply:
you kinda lost me here tbh :)  as stated; happy to put in whatever but I'm afraid I won't come up with better copy myself here :P

Reply:
ping @Wirone can you provide the required copy so this PR can go forwards?

Reply:
@SpacePossum just use "alternate" for unions (where only `|` is used) and DNF where actual DNF is used (mix of `|` and `&`) 😉.

Reply:
OK than I resolve this as I see no way how to improve this, thanks.

Reply:
Like [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7055/files#r1250732344) 🙄😅. As I said - `typed const alternate types 1` for `(A&B)|Z` is technically correct, BUT it's more precise if we explicitly name it as DNF (so we know there's union with intersections). It's also confusing that "alternate types" were tested for `TypeIntersectionTransformer`, while with DNF it's clear, because we know there's intersection too.